features/support/form_helpers.rb

```diff
@@ -336,11 +345,33 @@ def answer_all_service_questions(prompt_text)
       if find_elements_by_xpath("//input[@value='Mark as complete']").length > 0
         return
       end
-      next_answer_question_link = page.all(:xpath, "//a[text()='#{prompt_text}']")[0]
+
+      next_answer_question_link = find_elements_by_xpath("//a[text()='#{prompt_text}']")[0]
+
+      next_answer_question_link.click
+      # To debug particular section, use below line instead of above lines:
+      # find_elements_by_xpath("//span[text()='Section title here']/following::a[text()='Answer question']")[0].click
+      answer_service_section
+    end
+  end
+
+  def answer_all_dos_lot_questions(prompt_text)
+    i = 0
+
+    find_elements_by_xpath("//a[text()='#{prompt_text}']").length.times do
+      if find_elements_by_xpath("//input[@value='Mark as complete']").length > 0
+        return
+      end
+
+      next_answer_question_link = find_elements_by_xpath("//a[text()='#{prompt_text}']")[i]
+
+
       next_answer_question_link.click
       # To debug particular section, use below line instead of above lines:
       # find_elements_by_xpath("//span[text()='Section title here']/following::a[text()='Answer question']")[0].click
       answer_service_section
+
+      i += 1
```

Comment:
We could refactor above methods as follows to avoid repetition:

```
def is_service_complete
  find_elements_by_xpath("//input[@value='Mark as complete']").length > 0
end

def find_and_answer_next_question(prompt_text, index=0)
  next_answer_question_link = find_elements_by_xpath("//a[text()='#{prompt_text}']")[index]
  next_answer_question_link.click
  
  # To debug particular section, use below line instead of above lines:
  # find_elements_by_xpath("//span[text()='Section title here']/following::a[text()='Answer question']")[0].click
  
  answer_service_section
end

def answer_all_service_questions(prompt_text)
    while find_elements_by_xpath("//a[text()='#{prompt_text}']").length > 0
      if is_service_complete
        return
      end
      find_and_answer_next_question(prompt_text)  
    end
  end

  def answer_all_dos_lot_questions(prompt_text)
    i = 0
    find_elements_by_xpath("//a[text()='#{prompt_text}']").length.times do
      if is_service_complete
        return
      end
      find_and_answer_next_question(prompt_text, i)
      i += 1
    end
  end
```

Reply:
Done 👍

Reply:
💃